src/nls/api.jl

```diff
@@ -388,25 +388,61 @@ function hess_op_residual!(nls::AbstractNLSModel, x::AbstractVector, i::Int, Hiv
   )
 end
 
-function obj(nls::AbstractNLSModel, x::AbstractVector)
+"""
+    f = obj(nls::AbstractNLSModel, x::AbstractVector, Fx::AbstractVector)
+
+Evaluate `f(x)`, the objective function of `nls::AbstractNLSModel`. `Fx` is overwritten with the value of the residual `F(x)`.
+"""
```

Comment:
```suggestion
"""
function obj end

```

Reply:
Does it change something to have the empty function?

Reply:
If you only have one docstring for a function with multiple methods, it recommended to add the docstring above the empty function.
In the REPL, it will display the docstring for the `obj` function without specific argument. 
```
julia> ?
help> obj
```

Reply:
Otherwise, you should add a docstring for all methods.

Reply:
I see; actually, there is an issue connected to this https://github.com/JuliaSmoothOptimizers/NLPModels.jl/issues/385

Reply:
See QRMumps.jl for example:
https://github.com/JuliaSmoothOptimizers/QRMumps.jl/blob/main/src/QRMumps.jl